internal/compatibility/proto/request.go

```diff
@@ -623,3 +623,16 @@ func RefreshWorkflowTasksRequest(r *shared.RefreshWorkflowTasksRequest) *apiv1.R
 	}
 	return &request
 }
+
+func RestartWorkflowExecutionRequest(r *shared.RestartWorkflowExecutionRequest) *apiv1.RestartWorkflowExecutionRequest {
+	if r == nil {
+		return nil
+	}
+	request := apiv1.RestartWorkflowExecutionRequest{
+		Domain:            r.GetDomain(),
+		WorkflowExecution: WorkflowExecution(r.GetWorkflowExecution()),
+		Identity:          r.GetIdentity(),
+	}
```

Comment:
seems like proto is missing the `reason` field?  is that worth adding in now so it happens all at once?

Reply:
ah, no, this proto API is missing a field that's in the thrift API.  we should fix that.

Reply:
Tackling this in https://github.com/uber/cadence-idl/pull/128

Reply:
thanks Steven, very nice catch! I updated to latest idl where you included the reason and updated the mapper.